### PR TITLE
doc: add multi-threading section

### PR DESCRIPTION
**Pull request by mhdawson**
Signed-off-by: Michael Dawson <mdawson@devrus.com>

### CODE REVIEW

File: docs/functional-components/multi-threading.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  execute in their own set of processes in order to enable better
+  scaling.
+
+* [WorkerThreads](https://nodejs.org/api/worker_threads.html)
+  may also be appropriate if your application
+  must run as a single process (for example a desktop application). In these
+  cases it is known that you cannot scale beyond the resources of the single
+  machine and it is often preferrable to have the application show up
+  as a single processes versus many individual processes.
+
```
Comment (bajtos) [inline docs/functional-components/multi-threading.md:87]: Our friends at nearForm have created a module with higher-level and easier-to-use API for dealing with worker threads: [piscina](https://github.com/piscinajs/piscina). Is it something to mention in this doc?

Comment (mhdawson) [reply docs/functional-components/multi-threading.md:87]: Great minds think alike, I already did on line 67

**Review by rolfl — APPROVED** [review]
LGTM.

**Comment by mhdawson** [comment]
We discussed in the last Ref Arch meeting and there were no additional comments so landing.

**Pull request merged**